lib/webapp.js

```diff
@@ -533,6 +533,7 @@ function WebApp(options){
   this.expressApp.use(cookieParser());
   this.expressApp.use(session({
     //TODO properly generate this secret
+    name: 'connect.sid.' + options.httpsPort,
```

Comment:
This is not 100%.... it's possible for someone to have enabled http and disabled https (though discouraged).
Instead, check if https ISNT being used. If it isn't, use http. If both are being used, you can still use https since you have to choose only 1.